packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -106,6 +116,29 @@ class RequestWrapper {
   }
 
   /**
+   * Provides a method of setting the cacheableResponseCheck after the
+   * `RequestWrapper` has already been constructed, but only if it wasn't set
+   * inside the constructor.
+   *
+   * @param {function} callback The function used to determine whether a
+   *        response is cacheable.
+   */
+  set cacheableResponseCheck(callback) {
+    if (!this._cacheableResponseCheck) {
+      this._cacheableResponseCheck = callback;
+    }
+  }
+
+  /**
+   * @private
+   * @return {function} The function used to determine whether a response is
+   *         cacheable.
+   */
+  get cacheableResponseCheck() {
+    return this._cacheableResponseCheck || this._defaultCacheableResponseCheck;
```

Comment:
Can this not be just set in the constructor? i.e. when will this dynamically be set / changed? And when it does, could we actually go with functions for this instead of magic here.

setCacheableResponseCheck() and performCacheableResponseCheck()